pkg/k8shandler/forwarding_test.go

```diff
@@ -114,371 +76,258 @@ var _ = Describe("Normalizing Forwarding", func() {
 					{
 						Name:       "aPipeline",
 						OutputRefs: []string{output.Name, otherOutput.Name},
-						SourceType: logging.LogSourceTypeApp,
+						InputRefs:  []string{logging.InputNameApplication},
 					},
 				},
 			}
 		})
 		Context("pipelines", func() {
 
-			It("should only include logsources if there is atleast one valid pipeline", func() {
-				request.ForwardingSpec.Pipelines = []logging.PipelineSpec{
+			It("should only include inputs if there is at least one valid pipeline", func() {
+				request.ForwarderSpec.Pipelines = []logging.PipelineSpec{
 					{
 						Name:       "aPipeline",
 						OutputRefs: []string{"someotherendpoint"},
-						SourceType: logging.LogSourceTypeApp,
+						InputRefs:  []string{logging.InputNameApplication},
 					},
 				}
-				normalizedForwardingSpec := request.normalizeLogForwarding(namespace, cluster)
-				Expect(len(normalizedForwardingSpec.Pipelines)).To(Equal(0), "Exp. all pipelines to be dropped")
-				Expect(request.ForwardingRequest.Status.LogSources).To(Equal([]logging.LogSourceType{}), "Exp. no log sources")
+				spec, status := request.normalizeForwarder()
+				Expect(spec.Pipelines).To(BeEmpty(), "Exp. all pipelines to be dropped")
+				Expect(status.Inputs).To(BeEmpty())
```

Comment:
The test description implies the opposite of this assertion. Is this an error or just confusing?

Reply:
Don't follow. That line could be more neatly expressed as To(BeEmpy()) but it's still testing for emptiness - equality with an empty slice, "expect no sources". I started using some of Gomega's more expressive matchers, so the code is a little inconsistent in that regard but I think correct.

Reply:
Seems ok to me: it could be expressed more neatly as To(BeEmpty) but equal to an empty slice is the same thing. I started using some of the neater gomega assertions bud dint' clean up all the existing ones.